File: backend/src/models/SubtitleStream.js
const SubtitleStream = {
  title: {
    type: String,
    trim: true,
    default: "unknown",
  },

  index: {
    type: Number,
    required: [true, "index is required"],
  },

  codec: {
    type: String,
    required: [true, "codec is required"],
    trim: true,
  },

  language: {
    type: String,
    default: "unknown",
    trim: true,
  },
};

export default SubtitleStream;

File: backend/cmd/models/SubtitleStream.go
package models

import "gorm.io/gorm"

type Subtitles struct {
	gorm.Model
	Title    string `gorm:"not null;default:'unknown'" json:"title"`
	Index    int64  `json:"index"`
	Codec    string `gorm:"not null;default:'unknown'" json:"codec"`
	Language string `gorm:"not null;default:'unknown'" json:"language"`
	MovieID  uint   `json:"movieID"`
	Movie    Movie  `json:"movie"`
}
